src/context/TaskContext.tsx

```diff
@@ -0,0 +1,100 @@
+import { createContext, useContext, ReactNode } from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import taskService from "../services/taskService";
+import { type TaskContextProps, type Task } from "../types";
+
+import { Box, LinearProgress } from "@mui/material";
+import { notifySuccess, notifyError } from "../utils/notification";
+
+const TaskContext = createContext<TaskContextProps | undefined>(undefined);
+
+export const TaskProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const queryClient = useQueryClient();
+
+  const {
+    data: tasks = [],
+    isLoading,
+    isError,
+  } = useQuery<Task[]>("tasks", taskService.getTasks);
+
+  const addTaskMutation = useMutation(taskService.addTask, {
+    onSuccess: () => {
```

Comment:
Add queries and mutation in context if its needed to be used in multiple components in different heirarchy